From c9fb95a165df38b4d9ebe00f579549f9133aad73 Mon Sep 17 00:00:00 2001
Message-ID: <c9fb95a165df38b4d9ebe00f579549f9133aad73.1713654571.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1713654571.git.sharlatanus@gmail.com>
References: <cover.1713654571.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Tue, 16 Apr 2024 15:11:44 +0100
Subject: [PATCH 10/47] gnu: python-asdf: Update to 3.2.0.

* gnu/packages/astronomy.scm (python-asdf): Update to 3.2.0.
[arguments] <#:phases>: Replace 'fix-tests-setup with
'patch-pyprojectct-toml phase and add one more substitution rule.
[propagated-inputs]: Remove python-asdf-unit-schemas.

Change-Id: I9e7a0282ed0a9eb82a20988762792e10ae33496b
---
 gnu/packages/astronomy.scm | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index 55561c98e7..b06638c49c 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -4418,13 +4418,13 @@ (define-public python-tweakwcs
 (define-public python-asdf
   (package
     (name "python-asdf")
-    (version "3.1.0")
+    (version "3.2.0")
     (source
      (origin
        (method url-fetch)
        (uri (pypi-uri "asdf" version))
        (sha256
-        (base32 "0fa6y3gmqc0y3nz0h68vq3a84pvx6gc5zp33wg8a4n9b4kipm464"))))
+        (base32 "1wj556g15gwp6ir5hg083l15sifdsf23giqkx0jbn4lgdwjffbgr"))))
     (build-system pyproject-build-system)
     (arguments
      (list
@@ -4432,12 +4432,15 @@ (define-public python-asdf
       #~(list "-n" "auto" "-p" "no:legacypath")
       #:phases
       #~(modify-phases %standard-phases
-          ;; ImportError: Error importing plugin " no:legacypath": No module
-          ;; named ' no:legacypath'
-          (add-before 'check 'fix-tests-setup
+          (add-after 'unpack 'patch-pypojrect-toml
             (lambda _
               (substitute* "pyproject.toml"
-                ((".*:legacypath.*") "")))))))
+                ;; ImportError: Error importing plugin " no:legacypath": No
+                ;; module named ' no:legacypath'
+                ((".*:legacypath.*") "")
+                ;; TypeError: Configuration.__init__() got an unexpected
+                ;; keyword argument 'version_file'
+                (("version_file = \"asdf/_version.py\"") "")))))))
     (native-inputs
      (list python-fsspec
            python-packaging
@@ -4451,7 +4454,6 @@ (define-public python-asdf
     (propagated-inputs
      (list python-asdf-standard
            python-asdf-transform-schemas
-           python-asdf-unit-schemas
            python-attrs ;; for vendorized jsonschema
            python-importlib-metadata
            python-jmespath
-- 
2.41.0

